### PR TITLE
the bladed flatcap

### PR DESCRIPTION
**Pull request by QuietlyWhisper**
## About the PR

Adds bladed flatcaps, and their construction graphs, organized into the weapons section.
It is an improvised "concealed" weapon meant to surprise targets!

## Why / Balance

It is mildy better than just using your fists because it is a crafted weapon, however it is significantly worse than crafting a shiv because this item is meant to be used as a gimmick, a surprise throwing weapon!

Beyond that, I thought it was a cool reference to [some cool show I keep seeing clips of.](https://youtu.be/v4odL3rZkZc?si=_0w1ggxrRnNjj8wn&t=158)

## Media

crafting
hewearsit
image
image

[the discord video link](https://discord.com/channels/310555209753690112/675078881425752124/1213587907936264242)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Whisper
- add: Glass shards can be added to flatcaps to create bladed flatcaps!

### CODE REVIEW

**Comment by QuietlyWhisper** [comment]
Tests pass on my end idk why they dont pass on github... the path is correct, the image name is correct...

File: Resources/Prototypes/Entities/Clothing/Head/hats.yml [hunk 1]
```diff
@@ -938,10 +941,13 @@
   name: brown flatcap
   description: Stupid clown! You made me look bad!
   components:
+  - type: Tag
+    tags:
+    - BrimFlatcapBrown
   - type: Sprite
-    sprite: Clothing/Head/Hats/brownflatcap.rsi
+    sprite: Clothing/Head/Hats/Brownflatcap.rsi
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Entities/Clothing/Head/hats.yml:948]:
```suggestion
    sprite: Clothing/Head/Hats/brownflatcap.rsi
```

File: Resources/Prototypes/Entities/Clothing/Head/hats.yml [hunk 2]
```diff
@@ -938,10 +941,13 @@
...
   - type: Clothing
-    sprite: Clothing/Head/Hats/brownflatcap.rsi
+    sprite: Clothing/Head/Hats/Brownflatcap.rsi
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Entities/Clothing/Head/hats.yml:950]:
```suggestion
    sprite: Clothing/Head/Hats/brownflatcap.rsi
```

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml [hunk 3]
```diff
@@ -130,6 +130,44 @@
...
+  id: BladedFlatcapBrown
+  name: brown flatcap
+  description: Stupid clown! You made me look bad! It has glass shards hidden in the brim.
+  components:
+  - type: Construction
+    graph: BladedFlatcapBrown
+  - type: Sprite
+    sprite: Clothing/Head/Hats/Brownflatcap.rsi
+  - type: Clothing
+    sprite: Clothing/Head/Hats/Brownflatcap.rsi
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml:169]:
```suggestion
    sprite: Clothing/Head/Hats/brownflatcap.rsi
  - type: Clothing
    sprite: Clothing/Head/Hats/brownflatcap.rsi
```

File: Resources/Prototypes/Recipes/Construction/Graphs/weapons/bladed_hats.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - tag: GlassShard
+        name: a glass shard
+        icon:
+          sprite: Objects/Materials/Shards/shard.rsi
+          state: shard1
+        doAfter: 1
+      - tag: BrimFlatcapBrown
+        name: a brown flatcap
+        icon:
+          sprite: Clothing/Head/Hats/Brownflatcap.rsi
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Recipes/Construction/Graphs/weapons/bladed_hats.yml:40]:
```suggestion
          sprite: Clothing/Head/Hats/brownflatcap.rsi
```

File: Resources/Prototypes/Recipes/Construction/weapons.yml [hunk 5]
```diff
@@ -1,3 +1,25 @@
...
+
+- type: construction
+  name: brown bladed flatcap
+  id: BladedFlatcapBrown
+  graph: BladedFlatcapBrown
+  startNode: start
+  targetNode: icon
+  category: construction-category-weapons
+  description: An inconspicuous hat with glass shards sewn into the brim.
+  icon: { sprite: Clothing/Head/Hats/Brownflatcap.rsi, state: icon }
```
Comment (DrSmugleaf) [inline Resources/Prototypes/Recipes/Construction/weapons.yml:20]:
```suggestion
  icon: { sprite: Clothing/Head/Hats/brownflatcap.rsi, state: icon }
```

**Review by DrSmugleaf — DISMISSED** [review]

**Comment by Hanzdegloker** [comment]
Turning my child into a weapon are we? Fair enough.

**Comment by deltanedas** [comment]
sigma compilation 49

**Comment by IProduceWidgets** [comment]
https://www.youtube.com/watch?v=pIwLJtqoxBs

**Comment by QuietlyWhisper** [comment]
> [some cool show I keep seeing clips of.](https://www.youtube.com/watch?v=pIwLJtqoxBs)

I was half expecting another throwing hat reference, a few people mentioned Oddjob and Speedwagon

**Comment by TurboTrackerss14** [comment]
Might I be able to persuade you to mix up the damage numbers to 
thrown: ~8 
and melee: ~10 

Throwing a floopy flatcap and praying the bladed edge hits really shouldn't do double the damage of an actually aimed "swing" imo

**Review by LankLTE — APPROVED** [review]
Thrown damage being higher is weird, but it's kind of funny and I don't see it being particularly broken either.

**Pull request merged**